src/models/converters/tests/data_decoded.rs

```diff
@@ -0,0 +1,150 @@
+        data_decoded: data_decoded.clone(),
+        settings_info: SettingsInfo::SwapOwner {
+            prev_owner: "0x0000000000000000000000000000000000000001".to_string(),
+            old_owner: "0xA3DAa0d9Ae02dAA17a664c232aDa1B739eF5ae8D".to_string(),
+            new_owner: "0xF2CeA96575d6b10f51d9aF3b10e3e4E5738aa6bd".to_string(),
+        },
+    };
+
+    let actual = DataDecoded::to_settings_info(&data_decoded);
+
+    assert_eq!(expected.settings_info, actual);
+}
+
+#[test]
+fn data_decoded_change_threshold_to_settings_info() {
+    let data_decoded = serde_json::from_str::<DataDecoded>(crate::json::DATA_DECODED_CHANGE_THRESHOLD).unwrap();
+
+    let expected = SettingsChange {
+        data_decoded: data_decoded.clone(),
+        settings_info: SettingsInfo::ChangeThreshold {
+            threshold: 2
+        },
+    };
+
+    let actual = DataDecoded::to_settings_info(&data_decoded);
+
+    assert_eq!(expected.settings_info, actual);
+}
+
+#[test]
+fn data_decoded_change_implementation_to_settings_info() {
+    let data_decoded = serde_json::from_str::<DataDecoded>(crate::json::DATA_DECODED_CHANGE_MASTER_COPY).unwrap();
+
+    let expected = SettingsChange {
+        data_decoded: data_decoded.clone(),
+        settings_info: SettingsInfo::ChangeImplementation {
+            implementation: "0xb6029EA3B2c51D09a50B53CA8012FeEB05bDa35A".to_string()
+        },
+    };
+
+    let actual = DataDecoded::to_settings_info(&data_decoded);
+
+    assert_eq!(expected.settings_info, actual);
+}
+
+#[test]
+fn data_decoded_enable_module_to_settings_info() {
+    let data_decoded = serde_json::from_str::<DataDecoded>(crate::json::DATA_DECODED_ENABLE_MODULE).unwrap();
+
+    let expected = SettingsChange {
+        data_decoded: data_decoded.clone(),
+        settings_info: SettingsInfo::EnableModule {
+            module: "0xF5dC3718EEbC5b003F1672A499F2ACBE77Ba790d".to_string()
+        },
+    };
+
+    let actual = DataDecoded::to_settings_info(&data_decoded);
+
+    assert_eq!(expected.settings_info, actual);
+}
+
+#[test]
+fn data_decoded_disable_module_to_settings_info() {
+    let data_decoded = serde_json::from_str::<DataDecoded>(crate::json::DATA_DECODED_DISABLE_MODULE).unwrap();
+
+    let expected = SettingsChange {
+        data_decoded: data_decoded.clone(),
+        settings_info: SettingsInfo::DisableModule {
+            prev_module: "0xF5dC3718EEbC5b003F1672A499F2ACBE77Ba790d".to_string(),
+            module: "0x25F73b24B866963B0e560fFF9bbA7908be0263E8".to_string(),
+        },
+    };
+
+    let actual = DataDecoded::to_settings_info(&data_decoded);
+
+    assert_eq!(expected.settings_info, actual);
+}
+
+#[test]
+fn data_decoded_unknown_to_settings_info() {
+    let data_decoded = serde_json::from_str::<DataDecoded>(crate::json::DATA_DECODED_EXEC_TRANSACTION_FROM_MODULE).unwrap();
+
+    let expected = SettingsChange {
+        data_decoded: data_decoded.clone(),
+        settings_info: SettingsInfo::Unknown,
```

Comment:
Would it be nicer to (or more idiomatic) to have this as an optional and if we don't know what it is we just return `null`?

Reply:
just as a side note: I am fine either way. If at all I would ask what the clients prefer ;)

Reply:
Maybe null is fine. I think I opened a ticket in the android repo where they should handle polymorphic types more gracefully, so I think it is on them how they handle "unknown"